frame/staking-rewards/src/lib.rs

```diff
@@ -670,38 +687,14 @@ pub mod pallet {
 			let mut rewards_pool =
 				RewardPools::<T>::try_get(pool_id).map_err(|_| Error::<T>::RewardsPoolNotFound)?;
 
-			let mut inner_rewards = rewards_pool.rewards.into_inner();
-			for (asset_id, reward) in inner_rewards.iter_mut() {
-				let inflation = stake.reductions.get(asset_id).cloned().unwrap_or_else(Zero::zero);
-				let claim = if rewards_pool.total_shares == Zero::zero() {
-					Zero::zero()
-				} else {
-					reward
-						.total_rewards
-						.safe_mul(&stake.share)?
-						.safe_div(&rewards_pool.total_shares)?
-						.safe_sub(&inflation)?
-				};
-				let claim = if early_unlock {
-					(Perbill::one() - stake.lock.unlock_penalty).mul_ceil(claim)
-				} else {
-					claim
-				};
-				let claim = sp_std::cmp::min(
-					claim,
-					reward.total_rewards.safe_sub(&reward.claimed_rewards)?,
-				);
-				reward.claimed_rewards = reward.claimed_rewards.safe_add(&claim)?;
-				T::Assets::transfer(
-					reward.asset_id,
-					&Self::pool_account_id(&pool_id),
-					&stake.owner,
-					claim,
-					keep_alive,
-				)?;
-			}
-			rewards_pool.rewards = Rewards::try_from(inner_rewards)
-				.expect("Conversion must work as it's the same data structure; qed;");
+			(rewards_pool, _) = Self::collect_rewards(
+				&pool_id,
+				rewards_pool,
+				stake.clone(),
+				early_unlock,
```

Comment:
I would rename this to `is_early_unlock` for clarity

Reply:
2089aabe55eb6311ea6ae4c323d9c3639a4baa0a